src/porepy/models/fluid_mass_balance.py

```diff
@@ -987,16 +991,29 @@ def before_nonlinear_iteration(self):
             vals = interface_darcy_flux[
                 interface_offsets[id] : interface_offsets[id + 1]
             ]
-            data[pp.PARAMETERS][self.mobility_keyword].update({"darcy_flux": vals})
+            update_dicts(vals, data)
 
-        # The well fluxes are so few that we evaluate them individually, instead of
-        # dealing with the slightly more complex code to evaluate them all at once.
-        for intf, data in self.mdg.interfaces(return_data=True, codim=2):
-            vals = self.equation_system.evaluate(self.well_flux([intf]))
-            data[pp.PARAMETERS][self.mobility_keyword].update({"darcy_flux": vals})
+        wells = self.mdg.interfaces(codim=2)
+        well_darcy_flux = self.equation_system.evaluate(self.well_flux(wells))
+        well_offsets = np.cumsum([0] + [intf.num_cells for intf in wells])
+        for id, intf in enumerate(wells):
+            # Update the data dictionary with the Darcy flux for the current interface.
+            data = self.mdg.interface_data(intf)
+            vals = well_darcy_flux[well_offsets[id] : well_offsets[id + 1]]
+            update_dicts(vals, data)
 
         super().before_nonlinear_iteration()
 
+    def darcy_flux_keywords(self) -> list[str]:
```

Comment:
This is quite close to the attribute self.darcy_keyword (more generally, the word keyword is overused, but that is beyond the scope here). Perhaps a prefix that signifies this is where to put postprocessed, or computed darcy fluxes?

Reply:
Suggestions? "computed_darcy_fluxes_keywords" is off the table.

Reply:
Do you think either darcy_flux_storage_keywords or darcy_flux_parameter_keywords is an improvement? I don't have better suggestions.

Reply:
darcy_flux_storage_keywords  is better.